packages/frontend/src/stdlib/analyses/reachability.tsx

```diff
@@ -0,0 +1,98 @@
+import type { DblModel, ReachabilityProblemData } from "catlog-wasm";
+import { createMemo } from "solid-js";
+import type { ModelAnalysisProps } from "../../analysis";
+import { type ColumnSchema, FixedTableEditor, createNumericalColumn } from "../../components";
+
+import type { ObjectDecl } from "../../model";
+import type { ModelAnalysisMeta } from "../../theory";
+
+import "./simulation.css";
+
+/** Configuration for a mass-action ODE analysis of a model. */
+export type ReachabilityContent = ReachabilityProblemData;
+
+type Simulator = (model: DblModel, data: ReachabilityContent) => boolean;
+
+/** Configure a mass-action ODE analysis for use with models of a theory. */
+export function configureReachability(options: {
+    id?: string;
+    name?: string;
+    description?: string;
+    help?: string;
+    simulate: Simulator;
+}): ModelAnalysisMeta<ReachabilityContent> {
+    const {
+        id = "reachability",
+        name = "Reachability model checking",
+        description = "Check a Reachability formula",
+        help = "reachability",
+        ...otherOptions
+    } = options;
+    return {
+        id,
+        name,
+        description,
+        help,
+        component: (props) => <Reachability title={name} {...otherOptions} {...props} />,
+        initialContent: () => ({ tokens: {}, forbidden: {} }),
+    };
+}
+
+/** Analyze a model using Reachability formula. */
+export function Reachability(
+    props: ModelAnalysisProps<ReachabilityContent> & {
+        simulate: Simulator;
+        title?: string;
+    },
+) {
+    const obDecls = createMemo<ObjectDecl[]>(() => {
+        return props.liveModel.formalJudgments().filter((jgmt) => jgmt.tag === "object");
+    }, []);
```

Comment:
Sorry for the churn, but as of #702, we're no longer directly accessing the formal judgments in the notebook, but rather using the elaborated model. This is necessary for analyses to work with compositional models once that feature lands.

Have a look at the diff to an analysis in #702 to see what you need to change or ask me if it's not clear.